tests/lib/c_lib/src/main.c

```diff
@@ -812,6 +812,145 @@ void test_strtoul(void)
+
+	ret = strtoll(str_abnormal, &stop, 0);
+	zassert_equal(ret, 0, "strtoll base = 0 failed");
+	zassert_true((strcmp(stop, "ABCDEFGH") == 0), "strtoll get stop failed");
+
+	char border1[] = "-9223372036854775808";
+	char border2[] = "+9223372036854775807";
+	char border3[] = "+9223372036854775806";
+	char border4[] = "922337203685477580000000";
+
+	ret = strtoll(border1, NULL, 10);
+	zassert_equal(ret, LLONG_MIN, "strtoll base = 10 failed");
+	ret = strtoll(border2, NULL, 10);
+	zassert_equal(ret, LLONG_MAX, "strtoll base = 10 failed");
+	ret = strtoll(border3, NULL, 10);
+	zassert_equal(ret, 9223372036854775806, "strtoll base = 10 failed");
+	ret = strtoll(border4, NULL, 10);
+	zassert_equal(ret, LLONG_MAX, "strtoll base = 10 failed");
+}
+
+/**
+ *
+ * @brief test strtoull function
+ *
+ * @see strtoull().
+ *
+ */
+void test_strtoull(void)
+{
+	static const char buf1[] = "+10379aegi";
+	static const char buf2[] = "   -10379aegi";
+	static const char buf3[] = "-010379aegi";
+	static const char buf4[] = "0x10379aegi";
+	static const char buf5[] = "0X10379aegi";
+	static const char buf6[] = "01037aegi";
+	static const char buf7[] = "1037aegi";
+	static const char buf8[] = "++1037aegi";
+	static const char buf9[] = "A1037aegi";
+	static const char buf10[] = "a1037aegi";
+	static const char str_normal[] = "-1011 This stopped it";
+	static const char str_abnormal[] = "ABCDEFGH";
+	char *stop = NULL;
+	unsigned long long ret;
+
+	/* test function strtoull() */
+	ret = strtoull(buf3, NULL, 8);
+	zassert_equal(ret, -543, "strtoull base = 8 failed");
+	ret = strtoull(buf1, NULL, 10);
+	zassert_equal(ret, 10379, "strtoull base = 10 failed");
+	ret = strtoull(buf2, NULL, 10);
+	zassert_equal(ret, -10379, "strtoull base = 10 failed");
+	ret = strtoull(buf4, NULL, 16);
+	zassert_equal(ret, 17004974, "strtoull base = 16 failed");
+	ret = strtoull(buf4, NULL, 0);
+	zassert_equal(ret, 17004974, "strtoull base = 16 failed");
+	ret = strtoull(buf5, NULL, 0);
+	zassert_equal(ret, 17004974, "strtoull base = 16 failed");
+	ret = strtoull(buf6, NULL, 0);
+	zassert_equal(ret, 543, "strtoull base = 8 failed");
+	ret = strtoull(buf7, NULL, 0);
+	zassert_equal(ret, 1037, "strtoull base = 10 failed");
+	ret = strtoull(buf8, NULL, 10);
+	zassert_not_equal(ret, 1037, "strtoull base = 10 failed");
+	ret = strtoull(buf9, NULL, 10);
+	zassert_not_equal(ret, 1037, "strtoull base = 10 failed");
+	ret = strtoull(buf10, NULL, 10);
+	zassert_not_equal(ret, 1037, "strtoull base = 10 failed");
+
+	ret = strtoull(str_normal, &stop, 10);
+	zassert_equal(ret, -1011, "strtoull base = 10 failed");
+	zassert_true((strcmp(stop, " This stopped it") == 0), "strtoull get stop failed");
+
+	ret = strtoull(str_abnormal, &stop, 0);
+	zassert_equal(ret, 0, "strtoull base = 0 failed");
+	zassert_true((strcmp(stop, "ABCDEFGH") == 0), "strtoull get stop failed");
+
+	char border1[] = "+18446744073709551615";
+	char border2[] = "-18446744073709551615000";
+	char border3[] = "+18446744073709551619";
```

Comment:
Suggest adding a couple more cases here to test a long string of digits but not a large number, and a simply very large number:
```
border[] = "0x0000000000000000000000000000000000001"; // should pass
border[] = "10000000000000000000000000000000000001"; // should fail (too big)
border[] = "-10000000000000000000000000000000000001"; // should fail (too big)
```

Reply:
added those cases to strtoll and strtoull test.